File: tests/core/tools/codac2_tests_serialization.cpp
/** 
 *  Codac tests
 * ----------------------------------------------------------------------------
 *  \date       2024
 *  \author     Simon Rohou
 *  \copyright  Copyright 2024 Codac Team
 *  \license    GNU Lesser General Public License (LGPL)
 */

#include <catch2/catch_test_macros.hpp>
#include <codac2_serialization.h>

using namespace std;
using namespace codac2;

TEST_CASE("Serialization")
{
  Vector v({2,3,6});

  {
    std::ofstream out("data.cdc", std::ios::binary);
    serialize(out, v);
    out.close();
  }

  {
    Vector v_deserialized;
    std::ifstream in("data.cdc", std::ios::binary);
    deserialize(in, v_deserialized);
    CHECK(v == v_deserialized);
    in.close();
  }
}

File: tests/core/tools/codac2_tests_serialization.py
#!/usr/bin/env python

#  Codac tests
# ----------------------------------------------------------------------------
#  \date       2024
#  \author     Simon Rohou
#  \copyright  Copyright 2024 Codac Team
#  \license    GNU Lesser General Public License (LGPL)

import unittest
from codac import *
import sys

class TestSerialization(unittest.TestCase):

  def test_serialization(self):

    v = Vector([2,3,6])

    with open("data.cdc", "wb") as f:
      serialize(f, v)

    v_deserialized = Vector(1)
    with open("data.cdc", "rb") as f:
      deserialize(f, v_deserialized)

    self.assertTrue(v == v_deserialized)

if __name__ ==  '__main__':
  unittest.main()